### DISCUSSION TITLE
[NavigationMenu] Is it possible to make multilevel nested menu?

### DISCUSSION THREAD

**Discussion by Jbmanllr** [radix-ui/primitives#1765] [Help]
## Question

Hello, the menu would be almost perfect if the sub menu component behavior wasn't so confusing, i can't get it to work..

Can i achieve that : https://github.com/cdes/react-headless-nested-menu

if yes, how? Any examples?

Thanks !

**Comment by andy-hook** [comment]
@Jbmanllr This behaviour is supported out of the box with `DropdownMenu.Sub`, are you looking to use this specifically for website navigation?

**Reply by Jbmanllr** [reply]
Hi yes, i want to create a multi purpose responsive multi level nav menu. I pretty happy with what i have done with only conditional styling and viewport (mega menu/simple). But the sub menu component of the navigation menu is very confusing. Ideally i would also like to build faceted navigation (menu and drop down menu containing checkbox groups and radios)

**Reply by Jbmanllr** [reply]
> @Jbmanllr This behaviour is supported out of the box with `DropdownMenu.Sub`, are you looking to use this specifically for website navigation?

So should i use navigation root, then nest a drop down inside an nav menu item, then nest dropdowns inside each others? And ditch navigation Sub?

**Reply by Jbmanllr** [reply]
? :)

**Reply by andy-hook** [reply]
> And ditch navigation Sub?

`NavigationMenu.Sub` wasn't designed for this type of menu, [here's an example](https://601857eb614bae0021c9b9dd-vibchwvkwb.chromatic.com/?path=/story/components-navigationmenu--submenus) of the intention. I understand that's a bit confusing considering how the others behave!

I would suggest experimenting with the `DropdownMenu` as you mention. Depending on your design requirements I would personally try and optimise the navigation architecture such that you don't need this type of heavy nesting though.

**Reply by cpotey** [reply]
> `NavigationMenu.Sub` wasn't designed for this type of menu, [here's an example](https://601857eb614bae0021c9b9dd-vibchwvkwb.chromatic.com/?path=/story/components-navigationmenu--submenus) of the intention. I understand that's a bit confusing considering how the others behave!
> 
> I would suggest experimenting with the `DropdownMenu` as you mention. Depending on your design requirements I would personally try and optimise the navigation architecture such that you don't need this type of heavy nesting though.

Hi @andy-hook, are you able to share the JS code and CSS that created that your `NavigationMenu > Submenu` in the Storybook above at all? 
It's the sort of menu style I'm looking to create, but have been struggling to find "real world" examples for the `NavigationMenu.Sub` that include both a JS structure and accompanying styling/CSS

**Reply by joaom00** [reply]
Hi @cpotey, you can see the code here https://github.com/radix-ui/primitives/blob/main/packages/react/navigation-menu/src/NavigationMenu.stories.tsx#L186-L370

**Reply by cpotey** [reply]
> Hi @cpotey, you can see the code here https://github.com/radix-ui/primitives/blob/main/packages/react/navigation-menu/src/NavigationMenu.stories.tsx#L186-L370

@joaom00 Hero! Thank you muchly! ❤️
